### ISSUE TITLE
Denormalize some/all table using foreign keys

### ISSUE THREAD

**Issue by luis100** [keeps/db-preservation-toolkit#351]
Considering denormalizing into a single table.

> **Automatically denormalize tables using foreign keys.**
> There should be additional DBPTK module that produces "data warehouse" SIARD file where all the tables are aggregated to one big table using foreign keys.
> This feature is useful for smaller databases for researchers. More complex databases may need to have input list of tables of which tables should not be aggregated.

**Comment by Laurira** [comment]
Hi,

I started creating the use cases for this task and realized that there are lot of limitations due to the scalability. Big datasets can't be reasonably denormalized. Also I am not sure we want this only for the databases which store less than 10 tables.

Now we have better idea and vision how to make the usablity side better without denormalizing any tables but the idea serve the same goals. As #351 has considerable time estimation then lets substitute this task with new task as you suggested.

As most of the logic and queries are very often in the application side which will not reach into siard automatically then our idea is to pass the custom queries to the export process, so that DBPTK can create custom_query_views (which are materialized views) and hold the essential logic for the original usability side. These views should gain specific names so that they are strongly separated from the original database and user can understand that these are views that represent the logic of an application which used the original database.
You can suggest the implementations how to implement this task. Probably the queries need to be added via some file. Also the error reporting should be done before the main export process as these queries very often can be errorical.

**Comment by hmiguim** [comment]
Done in cc4a587.

###### To use the new feature:

When using a JDBC supported import module:
- `-icv <path>` or  `--import-custom-views=<path>` with a path to the query list file;

This file should follow the YAML syntax as follows:

```
schema_name:
    view_name: |
        query line 1
        query line 2
        ...
    another_view_name: |
        query line 1
        query line 2
        ...
```

**Comment by Laurira** [comment]
Can you improve the error handling. For example if user mistakes with sql in the yaml then the app should also produce the sql error message and tell that the dbptk did not reach to the end.

For example if I put false query to the yaml file (SELECT * FROM CUSTOMERS). Then app says "ERROR Fatal error while converting the database (Error getting custom view structure for sakila.minuVaade1)"

But it would be good if it says "Problems with custom query: select * from customers;	Error; Table 'sakila.customers' doesn't exist"

Then user can understand that the table customeR is the correct name of the table.
